### ISSUE TITLE
Modal customTheme not working

### ISSUE THREAD

**Issue by cristiangrojas** [themesberg/flowbite-react#1081]
## Steps to reproduce

I've the following implementation:

```
const customTheme: CustomFlowbiteTheme['modal'] = {
  content: {
    inner: 'relative rounded-lg bg-white shadow dark:bg-gray-800 flex flex-col max-h-[90vh]',
  },
  header: {
    base: 'hidden',
  },
  body: {
    base: 'hidden',
    popup: 'hidden'
  },
}

...

  return (
      <Modal
          dismissible
          show={isOpen}
          theme={customTheme}
          onClose={toggle}
      >

...
```

## Current behavior

It has no effect, but if I set styles globally like this:

```
const FlowbiteWrapper: FC<PropsWithChildren> = ({children}) => {
  const dark = localStorage.getItem('theme') === 'dark'

  return (
      <Flowbite theme={{ dark, theme }}>
        <PersistFlowbiteThemeToLocalStorage/>
        {children}
      </Flowbite>
  )
}
```

It works properly!

## Expected behavior

Theme to be applied to this particular modal.

**Comment by rluders** [comment]
@cristiangrojas recently we made some improves to the theme persistency and also the theme propagation for some components. Do you mind updating it to `@beta` and checking if the problem persists?

**Comment by SutuSebastian** [comment]
Fix confirmed in latest [0.7.0](https://github.com/themesberg/flowbite-react/releases/tag/v0.7.0) version.
